__tests__/pages/webhooks/checkout-calculate-taxes.test.ts

```diff
@@ -1,25 +1,43 @@
+/** @jest-environment setup-polly-jest/jest-environment-node */
 
-import { fetchTaxes } from "../../../backend/taxjarApi";
-import { FetchTaxesLinePayload, ResponseTaxPayload } from "../../../backend/types";
-import { getTaxJarConfig } from "../../../backend/utils";
-import handler from "../../../pages/api/webhooks/checkout-calculate-taxes";
+import { setupPollyMiddleware, setupRecording } from "../../pollySetup";
+import { PollyServer } from "@pollyjs/core";
 import {
   dummyCheckoutPayload,
   dummyFetchTaxesPayload,
   dummyFetchTaxesResponse,
   mockRequest,
 } from "../../utils";
-
-jest.mock("../../../backend/taxjarApi");
-
-const mockedFetchTaxesForCheckout = <jest.Mock>fetchTaxes;
+import { ResponseTaxPayload } from "../../../backend/types";
 
 describe("api/webhooks/checkout-calculate-taxes", () => {
-  afterEach(() => {
-    jest.clearAllMocks();
+  beforeAll(() => {
+    process.env.TAXJAR_FROM_COUNTRY = "PL";
+    process.env.TAXJAR_FROM_ZIP = "50-601";
+    process.env.TAXJAR_FROM_STATE = "";
+    process.env.TAXJAR_FROM_CITY = "Wroclaw";
+    process.env.TAXJAR_FROM_STREET = "Teczowa 7";
+    process.env.TAXJAR_SANDBOX = "true";
+    process.env.TAXJAR_API_KEY = "dummy";
+  });
+  const context = setupRecording();
+  beforeEach(() => {
+    const server = context.polly.server;
+    setupPollyMiddleware(server as unknown as PollyServer);
+    jest.unmock("taxjar/dist/util/request");
+    jest.resetModules();
   });
 
   it("rejects when saleor domain is missing", async () => {
+    const post = jest.fn((url: any, params: any) => ({
+      dummyFetchTaxesResponse,
+    }));
+    jest.doMock("taxjar/dist/util/request", () => ({
+      __esModule: true,
+      default: jest.fn((_) => ({ post: post })),
+    }));
+    const calculateTaxes = require("../../../pages/api/webhooks/checkout-calculate-taxes");
```

Comment:
This could be simplified to:

```js
jest.spyOn("taxjar/dist/util/request", "default").mockImplementation(() => ({ post }))
```

to restore the non-mocked implementation:

```ts
beforeEach(() => {
  jest.mockRestore();
})
```

This way you don't need to worry about using imports at the top of the file (`jest.spyOn` is not hoisted, unlike `jest.mock`) and you can still use the actual implementation in other tests 🥳